core/deployment/src/main/java/io/quarkus/deployment/CodeGenContext.java

```diff
@@ -9,12 +9,14 @@
     private final Path outDir;
     private final Path workDir;
     private final Path inputDir;
+    private final boolean redirectIO;
 
-    public CodeGenContext(AppModel model, Path outDir, Path workDir, Path inputDir) {
+    public CodeGenContext(AppModel model, Path outDir, Path workDir, Path inputDir, boolean redirectIO) {
         this.model = model;
         this.outDir = outDir;
         this.workDir = workDir;
         this.inputDir = inputDir;
+        this.redirectIO = redirectIO;
```

Comment:
why `redirectIO` and not `inheritIO` ?

Reply:
At the beginning named it like that. But after updating the `ProcessUtil` class from: 

    public static Process launchProcess(final ProcessBuilder builder, final Optional<ProcessInheritIODisabled> processInheritIODisabled)

to have this method signature:

    public static Process launchProcess(final ProcessBuilder builder, final boolean shouldRedirectIO)

calling this method from the native build step wth a `ProcessInheritIODisabled` would have give something like :

    launch(p, !processInheritIODisabled.isPresent())

So I preferred moving the negation in the `launchProcess` method.